### DISCUSSION TITLE
Other storage hosts?

### DISCUSSION THREAD

**Discussion by fangmarks** [supabase/supabase#1059] [Questions] [answered]
I saw #982 and was wondering how this was meant?

Like, would this allow us to provide Supabase with the Credentials for one (or more) S3 Bucket(s) or similar and essentially have Supabase play the Admin / Middleware for it?

I really like the current stuff that's available, but the pricing for the Storage is a bit high for me personally, so having the possibility of using my already existing Wasabi Buckets, in some way, would be amazing (Wasabi is S3 Compatible already). 

(I don't wanna self-host supabase if I don't have to, just to switch the S3 provider, honestly)

**Comment by kiwicopple** [comment, marked as answer]
You can vote on more hosts here :) https://github.com/supabase/supabase/discussions/982

> just to switch the S3 provider

At the moment we are focused on the functionality/features, but "BYO Cloud" is something we want to build. We don't have a timeline because the team is swamped already, but as soon as it becomes a concrete item on the roadmap we will announce it 👍

**Reply by surjithctly** [reply]
Hey,

Can I connect my Aws s3 bucket to supabase storage? If yes, how?

If no, what’s the point of adding multiple connections like wasabi?

**Reply by kiwicopple** [reply]
> connect my Aws s3 bucket

Currently everything is managed by Supabase - this makes it 10x easier to develop the platform and and debug problems. In the future we have some sort of BYO Cloud solution.

> If no, what’s the point of adding multiple connections like wasabi?

Self hosting & improvements for the future :) We want to focus development efforts on platforms which are popular

**Reply by surjithctly** [reply]
Got it, The problem is 2 GB is too limited on the free plan and 100 GB is too much to get started with Pro. 

I wish if its bit more generous on the free plan, so that upgrading to the pro plan make sense (similar to database)

or is it possible to increase storage limit only?

**Reply by kiwicopple** [reply]
It's not possible to upgrade one of these only (yet). But if you have some ideas for pricing improvements just let us know. We are willing to adjust the plans (within reason)

**Reply by surjithctly** [reply]
@kiwicopple Thanks. I propose two options.

1. Increase the storage GB limit so the developers are comfortable playing around without worry about the limit. 
  Then once the project is in production, it will be easy to upgrade. 

2. Provide an option to go directly to the Pay-as-you-go plan without a monthly fee (with a higher price of course)
    eg: S3 charges $0.023 per GB while you can take a higher price of $0.05 / GB. 
    Same for the database: Instead of the current 0.125 per GB, can take $2/gb 
    
    So total monthly bill if I'm using 4 GB Database + 50 GB Storage = $8 + $2.5 = $10.5 / month
    
    Once I exceed the $25 amount, I will be able to switch to the regular plan to save more. 
    
 Make sense?

**Reply by surjithctly** [reply]
Here's how I envision the plans which are affordable for developers, startups & enterprises.  (edited on original pricing page)

![image](https://user-images.githubusercontent.com/1884712/115193079-89ad9c80-a0a0-11eb-9fdc-939eb2ec1f06.png)

**Reply by kiwicopple** [reply]
Very interesting approach @surjithctly - thanks. I'll share this with the team

**Reply by philohelp** [reply]
Great idea IMHO, would fit my needs too. I'd love to be able to pay like $7/month for a small but already consistent project.

**Reply by tonyhart7** [reply]
we dont have pay as you go ??? yet ?

its been 1 year, pay as you go is more comfortable vs current pro tier
